### PR TITLE
CDRIVER-4572 Log database name in command_failed events

### PR DESCRIPTION
**Pull request by adriandole**

### CODE REVIEW

File: src/libmongoc/src/mongoc/mongoc-apm.c [hunk 1]
```diff
@@ -203,6 +203,7 @@ mongoc_apm_command_succeeded_init (mongoc_apm_command_succeeded_t *event,
                                    int64_t duration,
                                    const bson_t *reply,
                                    const char *command_name,
+                                   const char *database_name,
```
Comment (eramongodb) [inline src/libmongoc/src/mongoc/mongoc-apm.c:206]: Missing corresponding `event->database_name = database_name;`.

File: src/libmongoc/src/mongoc/mongoc-apm.h
Comment (eramongodb) [inline src/libmongoc/src/mongoc/mongoc-apm.h:1]: Missing `mongoc_apm_command_succeeded_get_database_name`.

File: src/libmongoc/tests/test-mongoc-command-monitoring.c [hunk 2]
```diff
@@ -1074,8 +1081,9 @@ command_failed_reply_command_failed_cb (
    test =
       (cmd_failed_reply_test_t *) mongoc_apm_command_failed_get_context (event);
    test->failed_calls++;
-   bson_destroy (&test->reply);
+   test->db = bson_strdup (mongoc_apm_command_failed_get_database_name (event));
 
+   bson_destroy (&test->reply);
```
Comment (eramongodb) [inline src/libmongoc/tests/test-mongoc-command-monitoring.c:1086]:
```suggestion

   bson_free (test->db);
   test->db = bson_strdup (mongoc_apm_command_failed_get_database_name (event));

   bson_destroy (&test->reply);
```
Potential memory leak. Ensure any prior value is deallocated before potentially being overwritten.

File: src/libmongoc/tests/test-mongoc-command-monitoring.c [hunk 3]
```diff
@@ -504,6 +507,7 @@ test_op_id_failed_cb (const mongoc_apm_command_failed_t *event)
    test = (op_id_test_t *) mongoc_apm_command_failed_get_context (event);
    ids.request_id = mongoc_apm_command_failed_get_request_id (event);
    ids.op_id = mongoc_apm_command_failed_get_operation_id (event);
+   test->db = bson_strdup (mongoc_apm_command_failed_get_database_name (event));
```
Comment (eramongodb) [inline src/libmongoc/tests/test-mongoc-command-monitoring.c:510]:
```suggestion

   bson_free(test->db);
   test->db = bson_strdup (mongoc_apm_command_failed_get_database_name (event));
```
Detected memory leak. Ensure any prior value is deallocated before potentially being overwritten.

**Review by eramongodb — CHANGES_REQUESTED** [review]
Missing spec test sync with changes from https://github.com/mongodb/specifications/commit/e7ee829329400786e01279b4f37d4e440d1e9cfa (syncing the logging and unified spec invalid tests should be unnecessary since we don't currently test either). Add `find.json` to to `./src/libmongoc/tests/json/command_monitoring/unified` and update event monitoring handlers in `entity-map.c` to include database names for [failed](https://github.com/mongodb/mongo-c-driver/blob/acc9ea8f918611c6d76d6979d8d44d56dce5e97e/src/libmongoc/tests/unified/entity-map.c#L519) and [succeeded](https://github.com/mongodb/mongo-c-driver/blob/acc9ea8f918611c6d76d6979d8d44d56dce5e97e/src/libmongoc/tests/unified/entity-map.c#L544) events accordingly.

File: src/libmongoc/src/mongoc/mongoc-apm.h [hunk 4]
```diff
@@ -160,6 +160,9 @@ mongoc_apm_command_failed_get_duration (
 MONGOC_EXPORT (const char *)
 mongoc_apm_command_failed_get_command_name (
    const mongoc_apm_command_failed_t *event);
+MONGOC_EXPORT (const char *)
+mongoc_apm_command_failed_get_database_name (
+   const mongoc_apm_command_failed_t *event);
```
Comment (kevinAlbs) [inline src/libmongoc/src/mongoc/mongoc-apm.h:165]: Add doc pages for `mongoc_apm_command_failed_get_database_name` and the to-be-added `mongoc_apm_command_succeeded_get_database_name`. See `mongoc_apm_command_started_get_database_name.rst` for an example.

**Review by eramongodb — APPROVED** [review]

**Pull request merged**